aten/src/ATen/core/ScalarType.h

```diff
@@ -38,6 +39,28 @@ enum class ScalarType {
   NumOptions
 };
 
+static inline DataType scalarTypeToDataType(ScalarType scalar_type) {
+#define DEFINE_CASE(ctype,name,_) \
+  case ScalarType:: name : return caffe2::TypeMeta::Id<ctype>();
+
+  switch(scalar_type) {
+    AT_FORALL_SCALAR_TYPES(DEFINE_CASE)
+    case ScalarType::Undefined: return DataType::uninitialized();
+    default: AT_ERROR("Unrecognized Scalartype ", scalar_type, " (please report this error)");
+  }
+#undef DEFINE_CASE
+}
+
+static inline ScalarType dataTypeToScalarType(DataType dtype) {
+#define DEFINE_IF(ctype,name,_) \
+  if (dtype == caffe2::TypeMeta::Id<ctype>()) { \
+    return ScalarType:: name; \
+  }
+  AT_FORALL_SCALAR_TYPES(DEFINE_IF)
```

Comment:
won't this still be asymmetric if someone passes in DataType::unitialized()?  (i.e. you won't return ScalarType::Undefined).